### PR TITLE
Amend contribution guide

### LINKED ISSUE #404: Update contributing guide
It would be good to expand on the contributing guide with a section for new contributors to provide a bit more detail about:
 - how they can make their first few PRs easy to review
 - some issues they can avoid with the automated checks
 - what to expect after they have submitted their first PR, etc.

I'm happy to submit a PR to address this.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
In this PR, I add a section to explain what happens after a pull request is submitted. This information might be useful for new contributors. This fixes #404.

In the process of the submitting the PR, some words were picked up by the spelling workflow and added to the wordlist. These can be found here d65b5ef5911a45b06f36ab88d4ce88f40876ac04 and here 74670b1486065b62734fa562e06002b6407f2d3d.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -40,6 +40,20 @@ precommit::use_precommit()
...
+
+*   Usually, all the review conversations occur under the PR. The reviewer merges the PR when every issue has been resolved. It is good practice to response to each raised conversation with a commit pointing to the change made to fix the issue.
+
+*   When a PR is ready to be merged, you may be asked to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) on the `main` branch. You can do this by checking out your branch and running `git rebase main`. If it is successful, your commits will be placed on top of the commit history of `main` in preparation for a merge. A rebase might result in some merge conflicts. Make sure that they are resolved, then push your changes to your branch again.
+
+*   A number of issues can cause the Github checks to fail. It might be helpful to safeguard against them by doing the following:
+  *   Check that there are no linting issues by running `lintr::lint_package()`.
+  *   Run `devtoools::check()` to check for wider package issues like mismatching documentation, e.t.c.
+  *   (Optional) Turn on continuous integration with Github Actions on your forked repository.
```
Comment (sbfnk) [inline .github/CONTRIBUTING.md:56]:
```suggestion
  *   (Optional) Turn on continuous integration with Github Actions on your forked repository.
  
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -40,6 +40,20 @@ precommit::use_precommit()
...
+
+*   Usually, all the review conversations occur under the PR. The reviewer merges the PR when every issue has been resolved. It is good practice to response to each raised conversation with a commit pointing to the change made to fix the issue.
+
+*   When a PR is ready to be merged, you may be asked to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) on the `main` branch. You can do this by checking out your branch and running `git rebase main`. If it is successful, your commits will be placed on top of the commit history of `main` in preparation for a merge. A rebase might result in some merge conflicts. Make sure that they are resolved, then push your changes to your branch again.
+
+*   A number of issues can cause the Github checks to fail. It might be helpful to safeguard against them by doing the following:
+  *   Check that there are no linting issues by running `lintr::lint_package()`.
+  *   Run `devtoools::check()` to check for wider package issues like mismatching documentation, e.t.c.
```
Comment (sbfnk) [inline .github/CONTRIBUTING.md:55]:
```suggestion
  *   Run `devtoools::check()` to check for wider package issues like mismatching documentation, etc. (this currently requires a fair bit of time/computation).
```

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (jamesmbaazam) [reply .github/CONTRIBUTING.md:55]: Can't see what was changed here.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -40,6 +40,20 @@ precommit::use_precommit()
...
 
+#### What happens after submitting a PR?
+
+*   PRs are reviewed by the team before they are merged. The review process only begins after the continuous integration checks, which are manually triggered by the maintainer, have passed.
+
+*   The Github Actions checks currently take a while (about an hour), so it might be helpful to "watch" the repository and check your email for a notification when it's   all done.
+
+*   Usually, all the review conversations occur under the PR. The reviewer merges the PR when every issue has been resolved. It is good practice to response to each raised conversation with a commit pointing to the change made to fix the issue.
+
+*   When a PR is ready to be merged, you may be asked to [rebase](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) on the `main` branch. You can do this by checking out your branch and running `git rebase main`. If it is successful, your commits will be placed on top of the commit history of `main` in preparation for a merge. A rebase might result in some merge conflicts. Make sure that they are resolved, then push your changes to your branch again.
```
[the remark this answers is not in the document]
Comment (sbfnk) [reply .github/CONTRIBUTING.md:50]: Sounds good, I've updated it.

**Review by sbfnk — APPROVED** [review]

**Review by seabbs — COMMENTED** [review]
Note this doesn't update the dev version of the new file with the changes made.

**Pull request merged**